File: Lua/Love2D/UI Stuff/src/classes/Signal.lua
local Signal = { callbacks = {} }
Signal.__index = Signal

function Signal:new()
    local instance = setmetatable({}, Signal)

    instance.callbacks = {}

    return instance
end

function Signal:connect(callback)
    table.insert(self.callbacks, callback)
end

function Signal:emit(...)
    for _, callback in pairs(self.callbacks) do
        callback(...)
    end
end

return Signal

File: C-ish/Raylib/Conrad Clicker/src/signal.cpp
#include "signal.h"

Signal::Signal(std::function<void()> callback) {
    callbacks.push_back(callback);
}

void Signal::Fire() const {
    for (const std::function<void()>& callback : callbacks) {
        callback();
    }
}

void Signal::Connect(std::function<void()> callback) {
    callbacks.push_back(callback);
}